course_inventory/inventory.py

```diff
@@ -183,10 +242,17 @@ def pull_sis_section_data_from_udw(section_ids: Sequence[int], conn: connection)
 
 def run_course_inventory() -> Sequence[Dict[str, Union[ValidDataSourceName, pd.Timestamp]]]:
     logger.info("* run_course_inventory")
+
+    # Initialize DBCreator object
+    db_creator_obj = DBCreator(INVENTORY_DB, APPEND_TABLE_NAMES)
+
     logger.info('Making requests against the Canvas API')
 
+    # Gather new_term_data
+    new_term_df = gather_new_term_data_from_api(ACCOUNT_ID, TERM_IDS, db_creator_obj)
+
     # Gather course data
-    course_df = gather_course_data_from_api(ACCOUNT_ID, TERM_ID)
+    course_df = gather_course_data_from_api(ACCOUNT_ID, TERM_IDS)
```

Comment:
instead of changing the gather_course_data_from_api() function, how about adding the term loop here, and just keep the gather_course_data_from_api() definition the same as before?

Reply:
I thought about this a little bit. I could do it that way, but I guess I thought we need to combine all the results anyway and this way the filtering could be done all at once. If you'd like me to change it to the way you're suggesting, that's fine, please ask me to do so directly :)